src/transforms/operations.js

```diff
@@ -106,14 +105,27 @@ export function insertTextOperation(transform, path, offset, text, marks) {
 export function joinNodeOperation(transform, path, withPath) {
   const { state } = transform
   const { document } = state
-  const node = document.assertPath(path)
-  const offset = node.length
+  const node = document.assertPath(withPath)
 
-  const inverse = [{
-    type: 'split_node',
-    path: withPath,
-    offset,
-  }]
+  let inverse
+  if (node.kind === 'text') {
+    const offset = node.length
+
+    inverse = [{
+      type: 'split_node',
+      path: withPath,
+      offset,
+    }]
+  } else {
+    // The number of children after which we split
+    const count = node.nodes.count()
+
+    inverse = [{
+      type: 'split_node',
+      path: withPath,
+      count,
```

Comment:
Why change this to `count` here? It feels like it makes the `split_node` operation harder to reason about and standardize. Instead could we not calculate the offset of the nodes and use that?